azurerm/data_source_api_management_product.go

```diff
@@ -14,7 +14,7 @@ func dataSourceApiManagementProduct() *schema.Resource {
 		Read: dataSourceApiManagementProductRead,
 
 		Schema: map[string]*schema.Schema{
-			"product_id": azure.SchemaApiManagementProductDataSourceName(),
+			"product_id": azure.SchemaApiManagementChildDataSourceName(),
```

Comment:
Is there a reason the property name is `product_id` while function is calling it the `name`? 🤔

Reply:
this validation method is applicable to all of the `children` of the API Management Service (for want of a better phrase)

in this instance whilst it's effectively the `name` (ala other resources) - however the API names this `product_id` as such I believe it's worth matching this